js_modules/dagster-ui/packages/ui-core/src/graphql/schema.graphql

```diff
@@ -713,6 +713,12 @@ enum BackfillPolicyType {
 type AssetDependency {
   asset: AssetNode!
   inputName: String!
+  partitionMapping: PartitionMapping
+}
+
+type PartitionMapping {
+  name: String!
+  description: String!
```

Comment:
This looks great to me! I'll have to think about how we show this on a per-dependency basis on the asset details page, but I think this is a great solution.